app/src/main/java/org/simple/clinic/patient/PatientQueryModel.kt

```diff
@@ -0,0 +1,15 @@
+package org.simple.clinic.patient
+
+import android.arch.persistence.room.Embedded
+
+data class PatientQueryModel(
```

Comment:
This class was introduced because Room doesn't have in-built support for 1:N relationships and a single patient can have multiple phone numbers.

Reply:
I took a look at it, and I don't think `@Relation` is a good fit for this use case. If we were only fetching a single patient model by `UUID`, it would be a great fit. However, we fetch all patients with a particular `SyncStatus`.

The current implementation using `PatientQueryModel` will only access the database once. However, with the `@Relation` annotation, we'll end up accessing it `numberOfPatients + 1` times.

Reply:
Also, TIL that Room has Relation support.

Reply:
TODO: Add comment about the existence of this class.

Reply:
TODO: Also confirm that it uses multiple database calls.

Reply:
Confirmed that `@Relation` makes only a single extra query to fetch data from the database for all parents. However, this has other issues, primarily, it does not allow loading relations into properties via the constructor and needs a settable property.

This is fixed in the newest release of `Room`, but that's an alpha release(`2.1.0-alpha01`) and  also requires migration to AndroidX. This change will be done at a later point in time when we finish this migration.